### ISSUE TITLE
[Build] Fix Build setup for oneAPI backend on Windows

### ISSUE THREAD

**Issue by edwinsolisf** [arrayfire/arrayfire#3609]
Versions of Intel's oneAPI Basekit after 2023.2 have changed the CMake workflow for using SYCL with Intel's compiler. Therefore, currently building ArrayFire with CMake and the Visual Studio Generator is not possible.

Description
===========
* Additional details about the errors during the build:

    1. Setting the language of a source file to SYCL causes the Visual Studio Generator to ignore and not set any Build Step to it. The first errors caused by this are observed on the `sort_by_key_${SBK_TYPE}` targets required by target `afoneapi`

    2. When adding SYCL to a target, the DPL includes are by default not added to the include directories of the targets causing a include not found compilation error on the `#include <oneapi/dpl/algorithm>` lines.
* What do you suspect is causing the issue?/Fixes
    1. [According to Intel oneAPI DPC++ Compiler Reference](https://www.intel.com/content/www/us/en/docs/dpcpp-cpp-compiler/developer-guide-reference/2024-2/use-cmake-with-the-compiler.html), a `find_package(IntelSYCL REQUIRED)` and `add_sycl_to_target` lines are required in CMake to force a source file to be build with a SYCL compiler. Furthermore, a newer CMake version requisites must also be set.
    2. To fix this, a `find_package(oneDPL REQUIRED)` must be added for cmake to find the dpl package. Then, it must be added to the include directories of the target.
* Which steps in the (wiki)[https://github.com/arrayfire/arrayfire/wiki] failed: `cmake --build . --target afoneapi`
* What operating system and/or distro are you using?: Windows 10/11
* Versions of the packages related to this bug: oneAPI Basetoolkit >= 2023.2

**Comment by aaronsuydam** [comment]
Agreed. This is broken on my machine as well, even after diving into all the CMakeLists.txt (before I found this issue) I can't even get it to detect that i have the oneAPI basekit installed. 

I wasted probably 10 hours over the last few weeks trying to get this to compile. Can someone add a note to the docs that this is broken rn??
